### ISSUE TITLE
Related-links de-duplication key should not look at profiling attributes

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2200]
Samples here:

http://www.oxygenxml.com/forum/files/dupLinks.zip

There is a situation in which a topic with a single child topic gets included in two parts of a DITA Map, with different profiling attributes specified on the ancestors, something like this:

      <!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "map.dtd">
      <map>
       <title>DITA Topic Map</title>
       <topicref href="parent.dita" audience="expert">
        <topicref href="child.dita"/>
       </topicref>
       <topicref href="parent.dita">
        <topicref href="child.dita"/>
       </topicref>
      </map>

Then the DITA Map is published to XHTML without any DITAVAL filtering (or with particular filtering which allows all topicrefs to be included).
When you look inside the generated "parent.html" instead of one you will find two references to the ""child.html".

Looking inside the XSLT:

     DITA-OT2.x\xsl\common\related-links.xsl

at the two functions called "related-links:hideduplicates" and "related-links:link", they create the uniqueness key by also appending existing profiling attributes like platform, product, audience to it.
In my opinion, this should not be done. The filtering is probably done before this stage anyway.

**Comment by raducoravu** [comment]
The issue can be reproduced with both DOT 1.8 and 2.x.

**Comment by jelovirt** [comment]
:+1: I think profiling attributes should not be considered in creating unique link set.

**Comment by robander** [comment]
:+1: from me too. I'm pretty sure (maybe 90% sure) the original logic exists because there was an expectation that links would be presented differently based on those attributes. It's still a valid question - what happens when `audience="expert"` is flagged, if you present only one link, how do you keep the flag? But I think that's much less likely than the case here, where the current processing just gives an extra link.

**Comment by raducoravu** [comment]
Right, the original logic did not take into account specializations of profiling attributes anyway.
